src/core/fair_queue.cc

```diff
@@ -223,35 +231,26 @@ void fair_queue::unplug_class(class_id cid) noexcept {
     unplug_priority_class(*_priority_classes[cid]);
 }
 
-auto fair_queue::grab_pending_capacity(const fair_queue_entry& ent) noexcept -> grab_result {
-    _group.maybe_replenish_capacity(_group_replenish);
-
-    if (_group.capacity_deficiency(_pending->head)) {
-        return grab_result::pending;
-    }
-
-    capacity_t cap = ent._capacity;
-    if (cap > _pending->cap) {
-        return grab_result::cant_preempt;
+auto fair_queue::reap_pending_capacity(bool& our_turn_has_come) noexcept -> capacity_t {
+    capacity_t ready = 0;
+    if (_pending.cap) {
+        _group.maybe_replenish_capacity(_group_replenish);
+        capacity_t deficiency = _group.capacity_deficiency(_pending.head);
+        our_turn_has_come = deficiency <= _pending.cap;
+        if (deficiency < _pending.cap) {
```

Comment:
nit: Took me a while to understand that this if is equivalent to `if (our_turn_has_come)`